### ISSUE TITLE
infinite sliders of variable width are not working

### ISSUE THREAD

**Issue by acontreras89** [akiran/react-slick#961]
The problem seems to be in [the way we look for the dom element](https://github.com/akiran/react-slick/blob/master/src/mixins/trackHelper.js#L135-L136) corresponding to the current slide:

```js
targetSlideIndex = (spec.slideIndex + spec.slidesToShow);
targetSlide = ReactDOM.findDOMNode(spec.trackRef).childNodes[targetSlideIndex];
```

After mounting the component, `spec.slideIndex` is likely to be `0`, but `ReactDOM.findDOMNode(spec.trackRef).childNodes[1]` returns the dom element corresponding to the copy of the last slide because, in my case, two elements are cloned from the back to the head of the slides.

**Comment by laveesingh** [comment]
Could you give the exact snippet that you used? 
I've been trying that myself and I think the issue lies in the calculation of `infiniteCount`. [Here](https://github.com/akiran/react-slick/blob/master/src/track.js#L100)
I also went back to original [jquery slick library](https://github.com/kenwheeler/slick/blob/master/slick/slick.js#L2409) where I encountered the same problem.

**Comment by Wgil** [comment]
Hello guys,

I have been looking into this issue while I get used to the library itself. Do you have any advance on this?

I analyzed your thoughts but still unable to connect the dots. If you can explain more why you think the issue could be there, it would be useful.

For what I've seen, The computed styles are wrong. Actually, if you set on the computed styles of the developer tools, a width equals to the actual slide width, it works normally and you can see it working if you have slides with the same width.

By the way, do you know if this feature is broken because it is under development or it just stopped working? The readme file says that it's not working.

**Comment by acontreras89** [comment]
In my case, two elements are being cloned from the rear to the head of the slides, which is odd because one would be just enough.

Debugging a little bit, I ended where my original comment points. There's definitely a bug, since the selector doesn't return the node it is supposed to. However, I do not know whether the bug is where I first pointed out, or where @laveesingh suggests (both make sense), or in any other place.
